pkg/authorize/handler.go

```diff
@@ -146,12 +152,18 @@ func NewHandler(logger log.Logger, client *http.Client, endpoint *url.URL, tenan
 			}
 			tenant = fields[tenantKey]
 		}
+
+		logger = log.With(logger, "tenant", tenant)
+
 		if _, err := AgainstEndpoint(logger, client, endpoint, token, tenant, nil); err != nil {
 			level.Warn(logger).Log("msg", "unauthorized request made:", "err", err)
 			w.WriteHeader(http.StatusUnauthorized)
 			return
 		}
 
+		// Log the tenant for debugging purposes.
+		level.Info(logger).Log("msg", "authorized request")
```

Comment:
should this be debug I wonder? might be spammy at info?

Reply:
The idea here is to have the `tenant` being logged alongside the request log. Thinking about the incident we had a few weeks ago, I though it would be great to have this data available at Info level (as well as the clusterID). And I cannot get it easily at the request logger level. So I have put it here. But yes, might be spammy.
I'll put debug level as suggested.